python/ray/experimental/channel/local_channel.py

```diff
@@ -0,0 +1,54 @@
+import uuid
+from typing import Any
+
+import ray
+from ray.experimental.channel import ChannelContext
+from ray.experimental.channel.common import ChannelInterface
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class LocalChannel(ChannelInterface):
+    def __init__(
+        self,
+        actor_handle: ray.actor.ActorHandle,
+    ):
+        """
+        LocalChannel is a channel for communication between two tasks in the same
+        worker process. It writes data directly to the worker's serialization context
+        and reads data from the serialization context to avoid the serialization
+        overhead and the need for reading/writing from shared memory.
+        """
+
+        # TODO (kevin85421): Currently, if we don't pass `actor_handle` to
+        # `LocalChannel`, the actor will die due to the reference count of
+        # `actor_handle` is 0. We should fix this issue in the future.
+        self._actor_handle = actor_handle
+        self.channel_id = str(uuid.uuid4())
+
+    def ensure_registered_as_writer(self) -> None:
+        pass
+
+    def ensure_registered_as_reader(self) -> None:
+        pass
+
+    def __reduce__(self):
+        return LocalChannel, (self._actor_handle,)
+
+    def write(self, value: Any):
+        # Because both the reader and writer are in the same worker process,
+        # we can directly store the data in the context instead of storing
+        # it in the channel object. This reduces the serialization overhead of `value`.
+        ctx = ChannelContext.get_current().serialization_context
+        ctx.set_data(self.channel_id, value)
```

Comment:
I assume this just writes an object reference to the serialization context? (or in the case of a primitive value, the entire primitive value is written?)

Reply:
I am not sure whether I understand this correctly or not. We don't serialize the data because the reader and writer are in the same actor process.